kshape/core.py

```diff
@@ -4,6 +4,8 @@
 from numpy.random import randint
 from numpy.linalg import norm, eigh
 from numpy.fft import fft, ifft
+from sklearn.externals.joblib import Parallel,delayed
+from sklearn.utils import check_random_state
```

Comment:
This is a new dependency and should be documented/added to https://github.com/Mic92/kshape/blob/master/setup.py#L37

Reply:
It might be desirable to make it optional as it is not supported by pypy yet as for the not parallel version only `check_random_state` is required, which is just a wrapper around numpy random: https://github.com/scikit-learn/scikit-learn/blob/f0ab589f/sklearn/utils/validation.py#L617

Reply:
Added to setup

How could we make it optional?

Reply:
I don't think scikit is required at all. It just wraps `np.random.RandomState` and [joblib](https://github.com/joblib/joblib). Does this code benefit from `joblib`? Otherwise [multiprocessing](https://docs.python.org/3/library/multiprocessing.html) might be sufficient.

Reply:
I use the helper for embarrassingly parallelization from joblib.

Reply:
Got rid of the scikit reference for joblib. Will get rid of the scikit reference for random_state once we figure what to do there.